offchain-modules/packages/x/src/ckb/tx-helper/generator.ts

```diff
@@ -474,9 +481,62 @@ export class CkbTxGenerator extends CkbTxHelper {
         return outputs.set(outputs.size - 1, changeOutput);
       });
     }
+
+    const omniLockConfig = ForceBridgeCore.config.ckb.deps.omniLock;
+    if (
+      omniLockConfig &&
+      fromLockscript.code_hash === omniLockConfig.script.codeHash &&
+      fromLockscript.hash_type === omniLockConfig.script.hashType
+    ) {
+      txSkeleton = txSkeleton.update('cellDeps', (cellDeps) => {
+        return cellDeps.push({
+          out_point: {
+            tx_hash: omniLockConfig.cellDep.outPoint.txHash,
+            index: omniLockConfig.cellDep.outPoint.index,
+          },
+          dep_type: omniLockConfig.cellDep.depType,
+        });
+      });
+
+      const messageToSign = (() => {
+        const hasher = new utils.CKBHasher();
+        const rawTxHash = utils.ckbHash(
+          core.SerializeRawTransaction(normalizers.NormalizeRawTransaction(createTransactionFromSkeleton(txSkeleton))),
+        );
+        // serialized unsigned witness
+        const serializedWitness = core.SerializeWitnessArgs({
+          lock: new Reader(
+            '0x' +
+              '00'.repeat(
+                SerializeRcLockWitnessLock({
+                  signature: new Reader('0x' + '00'.repeat(65)),
+                }).byteLength,
+              ),
+          ),
+        });
+        hasher.update(rawTxHash);
+        const lengthBuffer = new ArrayBuffer(8);
+        const view = new DataView(lengthBuffer);
+        view.setBigUint64(0, BigInt(new Reader(serializedWitness).length()), true);
+
+        hasher.update(lengthBuffer);
+        hasher.update(serializedWitness);
+        return hasher.digestHex();
+      })();
+
+      txSkeleton = txSkeleton.update('signingEntries', (signingEntries) => {
+        return signingEntries.push({
+          type: 'witness_args_lock',
+          index: 0,
+          message: messageToSign,
+        });
+      });
+    }
+
     logger.debug(`txSkeleton: ${transactionSkeletonToJSON(txSkeleton)}`);
     logger.debug(`final fee: ${await this.calculateCapacityDiff(txSkeleton)}`);
-    return txSkeletonToRawTransactionToSign(txSkeleton);
+
+    return txSkeleton;
   }
 
   async lock(
```

Comment:
1. need remove pw-lock related code
2. need add omni-lock witness if senderLockscript === omniLockscript
3. need calculate signingEntry for user to sign

Reply:
This have already been modified and is under local-ci.